dev/com.ibm.ws.jaxrs.2.0.cdi/src/com/ibm/ws/jaxrs20/cdi/component/JaxRsFactoryImplicitBeanCDICustomizer.java

```diff
@@ -418,7 +418,10 @@ public void onPrepareProviderResource(BeanCustomizerContext context) {
                     //else report warning, keep using provider from rs: change to use RuntimeType.POJO
                     else {
                         p.setRuntimeType(RuntimeType.POJO);
-                        resourcesManagedbyCDI.remove(p.getProviderResourceClass());
+                        // if CDI Scope is REQUEST_SCOPE, report warning and no action: get provider from JAXRS
+                        if (!(scopeName.contains(JAXRSCDIConstants.REQUEST_SCOPE))) {
+                            resourcesManagedbyCDI.remove(p.getProviderResourceClass());
```

Comment:
I think this is not allowed by the spec - see section 11.2.3:
```
In a product that supports CDI, implementations MUST support the use of CDI-style Beans as root re- source classes, providers and Application subclasses. Providers and Application subclasses MUST be singletons or use application scope.
```

That second sentence makes me think that we should not be allowing providers to be set to Request scope - maybe instead of enabling this, we should be logging an error message...  In any case, the logProviderMismatch call should be included in the if block, but I think the logic is not quite right.

What we may need to do is add a behavior switch that enables providers to use request scope - and then use your new code path if that switch is enabled (it should be disabled by default to keep the current behavior).